spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/MetaStore.scala

```diff
@@ -0,0 +1,573 @@
+      TempDatabaseOrTableToBeDeletedName,
+      TempDatabaseOrTableToBeDeletedName,
+      Option(cluster))
+    removeTable(tempTableIdent)
+  }
+
+  override def setTableOption(tableIdent: TableIdent, key: String, value: String) : Unit = {
+    val updateQuery =
+      s"""
+      |UPDATE ${getMetaStoreTableFullName}
+      |SET options['$key'] = '$value'
+      |WHERE cluster_name = '${tableIdent.cluster.get}'
+      | AND keyspace_name = '${tableIdent.keyspace}'
+      | AND table_name = '${tableIdent.table}'
+    """.stripMargin.replaceAll("\n", " ")
+    metaStoreConn.withSessionDo {
+      session => session.execute(updateQuery)
+    }
+  }
+
+  override def removeTableOption(tableIdent: TableIdent, key: String) : Unit = {
+    val deleteQuery =
+      s"""
+      |DELETE options['$key']
+      |FROM ${getMetaStoreTableFullName}
+      |WHERE cluster_name = '${tableIdent.cluster.get}'
+      | AND keyspace_name = '${tableIdent.keyspace}'
+      | AND table_name = '${tableIdent.table}'
+    """.stripMargin.replaceAll("\n", " ")
+    metaStoreConn.withSessionDo {
+      session => session.execute(deleteQuery)
+    }
+  }
+
+  /** Set table schema */
+  override def setTableSchema(tableIdent: TableIdent, scheamJsonString: String) : Unit = {
+    try {
+      DataType.fromJson(scheamJsonString)
+    } catch {
+      case e: Exception => throw new RuntimeException(s"$scheamJsonString is in wrong schema json string format")
+    }
+
+    val updateQuery =
+        s"""
+      |UPDATE ${getMetaStoreTableFullName}
+      |SET schema_json = '$scheamJsonString'
+      |WHERE cluster_name = '${tableIdent.cluster.get}'
+      | AND keyspace_name = '${tableIdent.keyspace}'
+      | AND table_name = '${tableIdent.table}'
+    """.stripMargin.replaceAll("\n", " ")
+    metaStoreConn.withSessionDo {
+      session => session.execute(updateQuery)
+    }
+  }
+
+  /** Remove table schema */
+  override def removeTableSchema(tableIdent: TableIdent) : Unit = {
+    val deleteQuery =
+      s"""
+      |DELETE schema_json
+      |FROM ${getMetaStoreTableFullName}
+      |WHERE cluster_name = '${tableIdent.cluster.get}'
+      | AND keyspace_name = '${tableIdent.keyspace}'
+      | AND table_name = '${tableIdent.table}'
+    """.stripMargin.replaceAll("\n", " ")
+    metaStoreConn.withSessionDo {
+      session => session.execute(deleteQuery)
+    }
+  }
+  override def storeDatabase(database: String, cluster: Option[String]) : Unit = {
+    val databaseNames = getAllDatabasesFromMetastore(cluster).toSet
+    if (databaseNames.contains(database))
+      return
+
+    val insertQuery =
+      s"""
+      |INSERT INTO ${getMetaStoreTableFullName}
+      | (cluster_name, keyspace_name, table_name)
+      | values (?, ?, ?)
+    """.stripMargin.replaceAll("\n", " ")
+
+    val clusterName = cluster.getOrElse(sqlContext.getCluster)
+    metaStoreConn.withSessionDo {
+      session =>
+        val preparedStatement = session.prepare(insertQuery)
```

Comment:
Incorrect use of preparedStatement. 
This is not the critical path, so no need for prepared statements.
But even if it was on the critical path, statements must not be prepared per every query execution. 
Use simple  `session.execute(query, param1, param2, ...)`